packages/webpack-cli/lib/utils/arg-parser.js

```diff
@@ -90,6 +90,26 @@ const argParser = (options, args, argsOnly = false, name = '') => {
                 // a multiple argument parsing function
                 const multiArg = (value, previous = []) => previous.concat([value]);
                 parserInstance.option(flagsWithType, option.description, multiArg, option.defaultValue).action(() => {});
+            } else if (option.multipleType) {
+                // for options which accept multiple types like env
+                // so you can do `--env platform=staging --env production`
+                // { platform: "staging", production: true }
+                const multiArg = (value, previous = {}) => {
+                    const [allKeys, val] = value.split('=');
```

Comment:
May I suggest:

```javascript
const [allKeys, val] = value.split(/=(.+)/, 2);
```

This can catch edge cases where the value contains `=` as well. For instance, the value might be a Base64 payload with padding, or any other complex value. The above snip will capture everything after the first `=` as the value.

in other words:

- `--env a=b=c` should yield `{ key: "a", value: "b=c" }`

Reply:
Yep missed that, thanks for correcting 💯